cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/Extensions.kt

```diff
@@ -1231,3 +1291,12 @@ val Expression.isImported: Boolean
     get() {
         return this.importedFrom.isNotEmpty()
     }
+
+private fun <T> ArrayDeque<T>.removeIfFirst(element: T): Boolean {
```

Comment:
Isn't this feature already provided by `SimpleStack`? Is there a reason not to use that class?

Reply:
Well the functionality is actually already implemented by the deque so why use the simple stack?

Reply:
I switched to the simple stack